profiler/src/main/java/com/splunk/opentelemetry/profiler/BasicJfrRecordingFile.java

```diff
@@ -44,15 +48,25 @@ public boolean tryAdvance(Consumer<? super RecordedEvent> action) {
               action.accept(jfr.readEvent(file, path));
               return true;
             }
+            closeSafely(file);
```

Comment:
This will not be closed if this stream is never consumed. Should we open jfr file lazily? Should we require actually closing the stream and then close the file?

Reply:
I don't think there's really a use case for opening the stream and not consuming it fully.  But yes, if a user did chose to do that then they would be responsible for closing it. Should I make a comment to that end?

Reply:
What about actually changing the API/implementation to prevent misuse?

Reply:
How about you add the close handler to the stream (`Stream#onClose(Runnable)`) and use try-with-resources on it?

Reply:
I like the idea of adding an `Stream.onClose()` callback to close the underlying file, but that still pushes the responsibility out and requires the user to do more.  I guess I'll try it out and see how it feels...

Reply:
> that still pushes the responsibility out and requires the user to do more.

Yeah, that's one of the drawbacks.
The worst part is that you don't really know if you're supposed to close the stream or not unless it's described in the Javadoc... I wish there was a `CloseableStream` instead